src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -203,225 +167,119 @@ public HttpResponseMessage RotateDocumentPages(PostedDataEntity postedData)
+
+                    viewOptions.RotatePage(1, angle);
+                    viewer.View(viewOptions);
                 }
-                return Request.CreateResponse(HttpStatusCode.OK, rotatedPages);
+
+                // TODO: remove response object
+                return Request.CreateResponse(HttpStatusCode.OK, new object());
             }
-            catch (System.Exception ex)
+            catch (Exception ex)
             {
                 // set exception message
-                return Request.CreateResponse(HttpStatusCode.InternalServerError, new Resources().GenerateException(ex));
+                return Request.CreateResponse(HttpStatusCode.OK, new Resources().GenerateException(ex));
             }
         }
 
-        /// <summary>
-        /// Download curerntly viewed document
-        /// </summary>
-        /// <param name="path">Path of the document to download</param>
-        /// <returns>Document stream as attachement</returns>
-        [HttpGet]
-        [Route("downloadDocument")]
-        public HttpResponseMessage DownloadDocument(string path)
+        private Rotation GetAngleValue(int angle)
         {
-            if (!string.IsNullOrEmpty(path))
-            {
-                if (File.Exists(path))
-                {
-                    HttpResponseMessage response = new HttpResponseMessage(HttpStatusCode.OK);
-                    var fileStream = new FileStream(path, FileMode.Open);
-                    response.Content = new StreamContent(fileStream);
-                    response.Content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
-                    response.Content.Headers.ContentDisposition = new ContentDispositionHeaderValue("attachment");
-                    response.Content.Headers.ContentDisposition.FileName = Path.GetFileName(path);
-                    return response;
-                }
+            switch (angle) 
+            {
+                case 90:
+                    return Rotation.On90Degree;
+                case 180:
+                    return Rotation.On180Degree;
+                case 270:
+                    return Rotation.On270Degree;
+                case -90:
+                    return Rotation.On270Degree;
+                case -180:
+                    return Rotation.On180Degree;
+                case -270:
+                    return Rotation.On90Degree;
+                default:
+                    return Rotation.On90Degree;
             }
-            return new HttpResponseMessage(HttpStatusCode.NotFound);
         }
 
-        /// <summary>
-        /// Upload document
-        /// </summary>
-        /// <param name="postedData">Post data</param>
-        /// <returns>Uploaded document object</returns>
-        [HttpPost]
-        [Route("uploadDocument")]
-        public HttpResponseMessage UploadDocument()
+        private LoadDocumentEntity LoadDocument(PostedDataEntity postedData, bool loadAllPages)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'loadAllPages'.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5242614)